### DISCUSSION TITLE
✨ Confidence in results

### DISCUSSION THREAD

**Discussion by massimag** [policratus/pupyl#198] [Ideas]
Hello, is there a way to return the confidence of the returned images/results?
(I mean, the distance beteween the input image and the results)
I need to check if an image is present of not in the database, and I would like to set my threshold.

**Comment by policratus** [comment]
Hey @massimag, how's it going? Thanks for opening up this discussion.

In the current stable version [0.14.2](https://github.com/policratus/pupyl/releases/tag/v0.14.2) there's no direct way to return the distances between the query and database images (unless you go deep into the library classes). This is easy to add and will be on the next version `0.14.3`. The implementation roadmap and detail can be followed in #199.

Although returning confidence is something wanted and feasible, the reason why you suggest this @massimag is already solved. Please check the [facets](https://pupyl.readthedocs.io/en/v0.14.1/facets.html#pupyl.indexer.facets.Index.append) module, specifically the `append` method:

```python
append(tensor, check_unique=False)
```

It contains a logic where before adding a new image to the index a check for unicity can be performed (despite it's disabled by default).

It solves your problem of checking if an image is already indexed? Or you need this check also during a search?

**Comment by policratus** [comment]
Hey @massimag, how are you?

Your suggestion was just merged and is part of `pupyl` by now, starting in version `0.14.3`.

Thanks again!

**Reply by massimag** [reply]
That's great! Thanks!!!
